Retarus/retarus_modules/retarus_events_forwarder.py

```diff
@@ -0,0 +1,60 @@
+import uuid
+from datetime import datetime, timezone
+from queue import SimpleQueue
+from threading import Event, Thread
+
+import dateutil.parser
+from kafka import KafkaProducer
+from kafka.errors import KafkaTimeoutError
+from sekoia_automation.connector import Connector
+
+from retarus_modules.metrics import OUTCOMING_EVENTS
+
+
+class RetararusEventsForwarder(Thread):
+    def __init__(self, connector: Connector, queue: SimpleQueue):
+        super().__init__()
+        self.connector = connector
+        self.configuration = self.connector.configuration
+        self.kafka_producer = KafkaProducer(bootstrap_servers=self.configuration.kafka_url)
+        self.queue = queue
+        self._stop_event = Event()
+
+    @property
+    def is_running(self):
+        return not self._stop_event.is_set()
+
+    def send_message(self, *messages):
+        try:
+            for message in messages:
+                stub = {
+                    "@timestamp": dateutil.parser.parse(message["ts"]).isoformat(),
+                    "customer": {"intake_key": self.configuration.intake_key},
+                    "message": message,
+                    "event": {
+                        "id": str(uuid.uuid4()),
+                        "created": datetime.now(timezone.utc).isoformat(),
+                        "outcome": "failure",
+                    },
+                }
+                self.kafka_producer.send(
+                    topic=self.configuration.kafka_topic,
+                    value=stub,
+                )
+                self.kafka_producer.flush()
+        except KafkaTimeoutError as ex:
+            self.connector.log_exception(ex, "Error sending the log to Kafka")
+
+    def run(self):
+        while self.is_running or self.queue.qsize() > 0:
+            try:
+                events = self.next_batch(self.max_batch_size)
```

Comment:
I don' t see the `next_batch` method ...

Reply:
Copied and pasted and I forgot to remove it, done